backend/functions/src/api/resources/user-profile/user-profile.model.ts

```diff
@@ -1,11 +1,57 @@
-export interface UserProfile {
+import { IsAlpha, IsDefined, IsEmail, IsEnum, IsNotEmpty, IsOptional, Length } from 'class-validator';
+import { Gender } from '../../../constants';
+
+export class UserProfile {
+  @Length(8, 20)
+  @IsNotEmpty()
+  @IsDefined({ message: 'Username is missing' })
   username: string;
+
+  @IsEmail()
+  @IsDefined({ message: 'Email is missing' })
+  @IsNotEmpty()
   email: string;
+
+  @Length(3, 30)
+  @IsAlpha()
+  @IsDefined({ message: 'Name is missing' })
+  @IsNotEmpty()
+  name: string;
+
+  @Length(10, 13)
+  @IsOptional()
   phoneNumber?: string;
-  gender?: 'male' | 'female' | '';
-  name?: string;
+
+  @IsOptional()
+  @IsEnum(Gender, { message: 'Invalid Gender' })
+  gender?: Gender;
+
+  @Length(0, 150)
   bio?: string;
+
   createdAt?: Date;
   updatedAt?: Date;
   isDeleted?: boolean;
```

Comment:
Why are these 3 fields here? Is the user submitting them from the form? Do they need validation?

Remove these. This class is not meant to model a `UserProfile`. It is only created for validating the payload in the request. Do not add fields that are not being checked for validity.